### PR TITLE
Recycler Now Accepts All Trash

### PR DESCRIPTION
**Pull request by civilCornball**
## About the PR 
Makes objects tagged "Trash" by the "TrashOnEmpty" component be recycled by the Recycler.
In practice, this means that every object that a Trash Bag can contain is recyclable.

Has been tested, but might cause issues with objects that have the "Recyclable" tag (not component, tag) and NOT the "Trash" tag.

**Changelog**
:cl: civilCornball
- fix: Fixes the recycler not recycling some trash.

### CODE REVIEW

File: Content.Server/Recycling/RecyclerSystem.cs [hunk 1]
```diff
@@ -92,8 +92,8 @@ private void Recycle(RecyclerComponent component, EntityUid entity)
         {
             RecyclableComponent? recyclable = null;
 
-            // Can only recycle things that are recyclable... And also check the safety of the thing to recycle.
-            if (!_tags.HasTag(entity, "Recyclable") &&
+            // Can only recycle things that are trash or recyclable... And also check the safety of the thing to recycle.
+            if (!_tags.HasTag(entity, "Trash") &&
```
Comment (Elijahrane) [inline Content.Server/Recycling/RecyclerSystem.cs:96]: Won't this make it stop working on stuff with the recyclable tag? Should probably be using hasanytag

**Review by Elijahrane — CHANGES_REQUESTED** [review]

**Comment by EmoGarbage404** [comment]
was there not an issue opened to get rid of the tag and only use the component? seeing as they basically fulfill the same purpose.

File: Content.Server/Recycling/RecyclerSystem.cs (hunk 1 above)
Comment (civilCornball) [reply Content.Server/Recycling/RecyclerSystem.cs:96]: Yes. That’s what I said in the description.
“HasAnyTag” is useful, though - thanks.

**Comment by civilCornball** [comment]
I honestly didn’t see it, and I’m not experienced enough to do that.

**Comment by vulppine** [comment]
to be honest, it probably would be nicer to just add the Recyclable component to the trash items, so that way something can come out of it (e.g., plastic for things like empty chip bags), but i do get the idea where some objects are too small to have any sort of meaningful amount of material that can be harvested from it

**Comment by LordEclipse** [comment]
> to be honest, it probably would be nicer to just add the Recyclable component to the trash items, so that way something can come out of it (e.g., plastic for things like empty chip bags), but there is something to consider where some objects are too small to have any sort of meaningful amount of material that can be harvested from it

Would it be possible to have it act like a lathe, like the item is made of plastic, but it doesn't "recycle" until it has enough of the amount of the resource? Banking the resources until there is enough for it to actually be made.

**Comment by vulppine** [comment]
> > to be honest, it probably would be nicer to just add the Recyclable component to the trash items, so that way something can come out of it (e.g., plastic for things like empty chip bags), but there is something to consider where some objects are too small to have any sort of meaningful amount of material that can be harvested from it
> 
> Would it be possible to have it act like a lathe, like the item is made of plastic, but it doesn't "recycle" until it has enough of the amount of the resource? Banking the resources until there is enough for it to actually be made.

this would be pretty similar to how i've seen a recycler work in a version of SS13, where it only takes out whole numbers of material

**Comment by Peptide90** [comment]
This is the way I’ve described it needing to be before, Freeman agreed. Thinks like syringes should give back less than one as some things are less than one to begin with.

The recycler should accumulate materials till it hits an integer then spit it out, a down stream stacker would also be nice so you’re not losing your new materials.

**Comment by civilCornball** [comment]
Well yes, but those suggestions can be another PR. I’m fairly certain we already have the capability to recycle into items anyways.
This is just a fix for something that bothered me.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**